lua/neo-tree/sources/filesystem/lib/filter.lua

```diff
@@ -215,22 +206,41 @@ M.show_filter = function(state, search_as_you_type, fuzzy_finder_mode, use_fzy)
       end
     end, 100)
     restore_height()
-  end, { once = true })
+  end
+
+  input:on({ event.BufLeave, event.BufDelete }, close_input, { once = true })
+
+  input:map("i", "<C-w>", "<C-S-w>", { noremap = true })
 
   if fuzzy_finder_mode then
     local config = require("neo-tree").config
     for lhs, cmd_name in pairs(config.filesystem.window.fuzzy_finder_mappings) do
       local t = type(cmd_name)
       if t == "string" then
-        local cmd = cmds[cmd_name]
-        if cmd then
-          input:map("i", lhs, create_input_mapping_handle(cmd, state, scroll_padding), { noremap = true })
+        if cmd_name == "close" then
```

Comment:
I think it might make more sense to rename the close_input function and put it in the cmds local along with `move_cursor_up/down` like:

```lua
local cmds = {
  -- ...
  close = function()
    -- ...
  end
}
```

Then we can remove the extra nesting of if-else.